### PR TITLE
feat: handle the case of simple unions

### PR DESCRIPTION
**Pull request by untereiner**
ADD the possibility to return only the type when the union is explicit
Issue #625

### CODE REVIEW

File: fastavro/_validation_py.py [hunk 1]
```diff
@@ -331,6 +331,7 @@ def validate_union(datum, schema, named_schemas, parent_ns=None, raise_errors=Tr
     "map": validate_map,
     "union": validate_union,
     "error_union": validate_union,
+    "nullable_union": validate_union,
```
Comment (scottbelden) [inline fastavro/_validation_py.py:334]: This isn't needed anymore, right?

File: fastavro/utils.py [hunk 2]
```diff
@@ -60,7 +60,11 @@ def gen_data(schema: Schema, named_schemas: NamedSchemas, index: int) -> Any:
             _gen_utf8(): gen_data(map_schema["values"], named_schemas, index)
             for _ in range(10)
         }
-    elif record_type == "union" or record_type == "error_union":
```
Comment (scottbelden) [inline fastavro/utils.py:63]: This can be reverted back, right?

**Review by scottbelden — CHANGES_REQUESTED** [review]

File: tests/test_fastavro.py [hunk 3]
```diff
@@ -1985,6 +1985,43 @@ def test_return_record_name_with_named_type_in_union():
     assert records == rt_records
 
 
+def test_return_record_with_named_type_in_union():
```
Comment (scottbelden) [inline tests/test_fastavro.py:1988]: Can we add another test where there is a union of `null`, `record1`, and `record2` and someone sets `return_record_name` to `True`? The code as is looks like it would consider that a nullable union and override the `return_record_name` which we wouldn't want.

**Comment by scottbelden** [comment]
The intention here is to only handle simple unions (union of null and another record), right? It feels like we could extend this to be a union of any size as long as there is only one record, but perhaps that's best left out for now since I believe it should be possible to expand to that without a breaking change.

File: fastavro/_validation_py.py (hunk 1 above)
Comment (untereiner) [reply fastavro/_validation_py.py:334]: yes, my mistake

File: fastavro/utils.py (hunk 2 above)
Comment (untereiner) [reply fastavro/utils.py:63]: yep

File: tests/test_fastavro.py (hunk 3 above)
Comment (untereiner) [reply tests/test_fastavro.py:1988]: I believe you are right. I could add a len == 2 in the is_nullable test

**Comment by untereiner** [comment]
It is the exact intent. I don't know yet if it is extendable to unions of any size of simple types + one record. I try something

**Comment by untereiner** [comment]
I juste have a doubt about the test: `if extracted_type not in AVRO_TYPES or extracted_type == "record":`
The "record" type is also in AVRO_TYPES. 

Should AVRO_TYPES not be separated into primitives types and complex types ? 
Knowing that this simple union thing works only with primitive types I guess ?

**Comment by untereiner** [comment]
I didn't think so far. I thought it wouldn't work. But yes if it works I would expect to get the value and not a tuple.
For me it is all good! 
Thanks for working on it !
